test/unit/config.test.ts

```diff
@@ -31,50 +32,91 @@ function createExplorer(overrides: Partial<ReturnType<typeof cosmiconfig>> = {})
   };
 }
 
-suite('config', function () {
-  test('loadConfig() returns undefined when no config exists', async function () {
+suite('loadConfig()', function () {
+  test('returns an Result Err when no config exists', async function () {
     const explorer = createExplorer({
-      load: sinon.fake.rejects(new Error()),
+      load: sinon.fake.rejects(new Error('Not found')),
     });
-    const actual = await loadConfig('./not-found.js', explorer);
-    const expected = undefined;
+    const config = await loadConfig('./not-found.js', explorer);
+    if (config.isOk()) {
+      assert.fail('loadConfig() did not returned an expected error');
```

Comment:
You could avoid the `if` statement by using `assert.strictEqual(config.isOk(), false, 'loadConfig() did not returned an expected error')`.

Reply:
Unfortunately not. TypeScript's control flow analysis is not smart enough here and only works with an `if` statement 😕 So with this "inline" assertion the compiler still thinks that property 'error' does not exist.

Reply:
Hopefully this will be fixed in the next version of typescript.